### PR TITLE
store: use range requests if we have a local file already

### PR DESCRIPTION
**Pull request by mvo5**

### CODE REVIEW

File: store/store.go [hunk 1]
```diff
@@ -1124,7 +1147,7 @@ var download = func(name, sha3_384, downloadURL string, user *auth.UserState, s
 		// connections) that led us to an error (the default client is
 		// documented as not reusing the transport unless the body is
 		// read to EOF and closed, so this is a belt-and-braces thing).
-		r, err := s.doRequest(&http.Client{}, reqOptions, user)
+		r, err := s.doRequest(newHTTPClient(0), reqOptions, user)
```
Comment (chipaca) [inline store/store.go]: this means using a logging transport for download, which means that if the transport gets `debugBody` we'll now have gigabytes worth of debug logs.
Easy fix though: in `store/logger.go`, add a content-type arg to the `debugBody()` method, and only return true if the content type is not `application/octet-stream`. Pass in the request/response content type when calling it from `RoundTrip`.
I'll poke you on telegram to see what your load is like; I can do these changes in a branch of my own.

File: store/store.go [hunk 2]
```diff
@@ -1072,7 +1072,8 @@ func (s *Store) Download(name string, targetFn string, downloadInfo *snap.Downlo
 	if err := os.MkdirAll(filepath.Dir(targetFn), 0755); err != nil {
 		return err
 	}
-	w, err := os.Create(targetFn + ".partial")
+	resume := int64(0)
+	w, err := os.OpenFile(targetFn+".partial", os.O_APPEND|os.O_RDWR|os.O_CREATE, 0644)
```
Comment (mvo5) [inline store/store.go]: Thanks @zyga, this is indeed a bit dangerous, I changed it to use seek-after-open now.

File: store/store.go (hunk 1 above)
Comment (mvo5) [inline store/store.go]: Actually after merging your branch I think its fine if we make it an explicit option (like you did there). Please let me know if you prefer the auto-detect approach, both is fine with me, I'm slightly on the fence but lean towards explicit.

**Comment by mvo5** [comment]
@zyga thanks, I can't find the question, happy to answer it when I find it.

**Pull request merged**